Core/src/Geometry/CuboidVolumeStack.cpp

```diff
@@ -0,0 +1,893 @@
+    Volume::update(std::make_shared<CuboidVolumeBounds>(*bounds), std::nullopt,
+                   logger);
+    ACTS_VERBOSE("Transform is now: " << m_transform.matrix());
+    return;
+  }
+
+  ACTS_VERBOSE("Checking volume alignment");
+  checkVolumeAlignment(volumeTuples, logger);
+
+  auto dirIdx = static_cast<std::size_t>(m_dir);
+  ACTS_VERBOSE("Sorting by volume " << axisDirectionName(m_dir) << " position");
+  std::ranges::sort(volumeTuples, {}, [&dirIdx](const auto& v) {
+    return v.localTransform.translation()[dirIdx];
+  });
+  ACTS_VERBOSE("Checking for overlaps and attaching volumes in "
+               << axisDirectionName(m_dir));
+  std::vector<VolumeTuple> gapVolumes =
+      checkOverlapAndAttach(volumeTuples, strategy, logger);
+
+  ACTS_VERBOSE("Appending " << gapVolumes.size()
+                            << " gap volumes to the end of the volume vector");
+  std::copy(gapVolumes.begin(), gapVolumes.end(),
+            std::back_inserter(volumeTuples));
+
+  ACTS_VERBOSE("*** Volume configuration after " << axisDirectionName(m_dir)
+                                                 << " attachment:");
+  printVolumeSequence(volumeTuples, logger, Acts::Logging::VERBOSE);
+
+  ACTS_VERBOSE("Synchronizing bounds in " << axisDirectionName(m_dirOrth1)
+                                          << "/"
+                                          << axisDirectionName(m_dirOrth2));
+  const auto [hl1, hl2] = synchronizeBounds(volumeTuples, logger);
+
+  for (auto& vt : volumeTuples) {
+    ACTS_VERBOSE("Updated bounds for volume at "
+                 << axisDirectionName(m_dir) << ": "
+                 << vt.localTransform.translation()[dirIdx]);
+    ACTS_VERBOSE(*vt.updatedBounds);
+
+    vt.commit(logger);
+  }
+
+  ACTS_VERBOSE("*** Volume configuration after "
+               << axisDirectionName(m_dirOrth1) << "/"
+               << axisDirectionName(m_dirOrth2) << " synchronization:");
+  printVolumeSequence(volumeTuples, logger, Acts::Logging::VERBOSE);
+
+  std::ranges::sort(volumeTuples, {},
+                    [*this](const auto& v) { return v.mid(m_dir); });
+
+  m_volumes.clear();
+  for (const auto& vt : volumeTuples) {
+    m_volumes.push_back(vt.volume);
+  }
+
+  ACTS_DEBUG("*** Volume configuration after final " << axisDirectionName(m_dir)
+                                                     << " sorting:");
+  printVolumeSequence(volumeTuples, logger, Acts::Logging::DEBUG);
+
+  double min = volumeTuples.front().min(m_dir);
+  double max = volumeTuples.back().max(m_dir);
+
+  double mid = (min + max) / 2.0;
+  double hl = (max - min) / 2.0;
+
+  Translation3 translation(Vector3::Unit(dirIdx) * mid);
+  m_transform = m_groupTransform * translation;
+  std::array<double, 3> bounds;
+  bounds.at(dirIdx) = hl;
+  bounds.at(static_cast<std::size_t>(m_dirOrth1)) = hl1;
+  bounds.at(static_cast<std::size_t>(m_dirOrth2)) = hl2;
+
+  Volume::update(std::make_shared<CuboidVolumeBounds>(bounds), std::nullopt,
```

Comment:
Could you use the `CuboidVolumeBounds` constructor that uses named arguments? We're trying to phase out directly constructing from the "bound values" vectors / arrays, since this is tied to the internal representation.